tests/Module/Checkout/RateCalculatorTest.php

```diff
@@ -0,0 +1,251 @@
+<?php
+
+declare( strict_types=1 );
+
+namespace Tests\Module\Checkout;
+
+use Packetery\Module\Carrier\Options;
+use Packetery\Module\Checkout\RateCalculator;
+use Packetery\Module\Framework\WcAdapter;
+use Packetery\Module\Framework\WpAdapter;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Tests\Module\CarrierOptionsDummyFactory;
+use Tests\Module\MockFactory;
+
+class RateCalculatorTest extends TestCase {
+
+	private WpAdapter|MockObject $wpAdapter;
+	private RateCalculator $rateCalculator;
+
+	private function createRateCalculatorMock(): void {
+		$this->wpAdapter = MockFactory::createWpAdapter( $this );
+
+		$this->rateCalculator = new RateCalculator(
+			$this->wpAdapter,
+			$this->createMock( WcAdapter::class ),
+			MockFactory::createCurrencySwitcherFacade( $this ),
+		);
+	}
+
+	/**
+	 * @return array
+	 */
+	public static function calculationDataProvider(): array {
+		return [
+			[
+				'expectedCost'      => 11,
```

Comment:
měla by asi být float :) to same taotalProductValue a cartWeightKG.. neboo se pletu?